### PR TITLE
Add micro-ROS Custom types tutorial

### PR DESCRIPTION
**Pull request by pablogs9**
This PR adds micro-ROS custom types tutorial

### CODE REVIEW

File: docs/rst/tutorials/micro/custom_types/custom_types.rst [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+
+.. contents:: Table of Contents
+    :depth: 1
+    :local:
+    :backlinks: none
+
+
+This tutorial aims at providing step-by-step guidance for those users interested in adding custom message definition to the micro-ROS build. Those instructions can be used to extend the type definition for topics, services and actions provided by ROS 2 and micro-ROS.
+
+.. TODO(pgarrido): explain how to integrate in build systems
```
Comment (dcoello-eProsima) [inline docs/rst/tutorials/micro/custom_types/custom_types.rst:14]: can be done on this PR, or is it going to be in a new tutorial? is it the line 21 link?

File: docs/rst/tutorials/micro/custom_types/custom_types.rst [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+  msg.point32_test.z = 3.12;
+
+  ...
+
+.. TODO(pgarrido): add link here to memory handling tutorial
+
+
+
+
+.. The micro-ROS middleware, **eProsima Micro XRCE-DDS**, provides a user API that allows interfacing with the lowest level transport layer at runtime, which enables users to implement their own transports in both the micro-ROS Client and micro-ROS Agent libraries.
```
Comment (dcoello-eProsima) [inline docs/rst/tutorials/micro/custom_types/custom_types.rst:166]: dead code?

File: docs/rst/tutorials/micro/custom_types/custom_types.rst (hunk 1 above)
Comment (pablogs9) [reply docs/rst/tutorials/micro/custom_types/custom_types.rst:14]: aa6c6df32fcb0e537b016fdfa4e81f437e518598

Comment (pablogs9) [reply docs/rst/tutorials/micro/custom_types/custom_types.rst]: aa6c6df32fcb0e537b016fdfa4e81f437e518598

File: docs/rst/tutorials/micro/custom_types/custom_types.rst (hunk 2 above)
Comment (pablogs9) [reply docs/rst/tutorials/micro/custom_types/custom_types.rst]: aa6c6df32fcb0e537b016fdfa4e81f437e518598

**Review by dcoello-eProsima — APPROVED** [review]

File: docs/rst/tutorials/micro/custom_types/custom_types.rst [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    :backlinks: none
+
+
+This tutorial aims at providing step-by-step guidance for those users interested in adding custom message definition to the micro-ROS build. Those instructions can be used to extend the type definition for topics, services and actions provided by ROS 2 and micro-ROS.
+
+.. note::
+
+    This tutorial starts from a previously created micro-ROS environment. Check the first steps of `Getting started micro-ROS <https://docs.vulcanexus.org/en/latest/rst/tutorials/micro/getting_started/getting_started.html>`_ for instructions on how to create a micro-ROS environment for embedded platforms.
+
+    For creating custom types in a :ref:`Build System Component <tutorials_micro_build_system_components>`, check the instructions in each components' repository. For example `Renesas e2 Studio Readme <https://github.com/micro-ROS/micro_ros_renesas2estudio_component#adding-custom-packages-to-the-micro-ros-build>`_ or `ST Micro ST Cube IDE/MX <https://github.com/micro-ROS/micro_ros_stm32cubemx_utils#adding-custom-packages>`_
```
Comment (Acuadros95) [inline docs/rst/tutorials/micro/custom_types/custom_types.rst:18]:
```suggestion
    For creating custom types in a :ref:`Build System Component <tutorials_micro_build_system_components>`, check the instructions in each component repository. For example `Renesas e2 Studio Readme <https://github.com/micro-ROS/micro_ros_renesas2estudio_component#adding-custom-packages-to-the-micro-ros-build>`_ or `ST Micro ST Cube IDE/MX <https://github.com/micro-ROS/micro_ros_stm32cubemx_utils#adding-custom-packages>`_.
```

File: docs/rst/tutorials/micro/custom_types/custom_types.rst [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+  ...
+
+  rclc_publisher_init_default(&publisher, &node, ROSIDL_GET_MSG_TYPE_SUPPORT(my_custom_message, msg, MyCustomMessage), "my_custom_publisher");
+  rcl_publish(&publisher, &msg, NULL);
+
+  ...
+
+
+Further information can be found in :ref:`Implementing custom interfaces <SinglePkgInterface>`.
```
Comment (Acuadros95) [inline docs/rst/tutorials/micro/custom_types/custom_types.rst:98]: I would move this to the start of the page

File: docs/rst/tutorials/micro/custom_types/custom_types.rst [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+Type composition
+----------------
+
+It is possible to create custom types that compose members from another ROS 2 message type packages. For example a member with type ``Point32`` from the ROS 2 package ``geometry_msgs``.
+
+First of all, the dependency shall be included in  in the ``CMakeLists.txt``:
```
Comment (Acuadros95) [inline docs/rst/tutorials/micro/custom_types/custom_types.rst:105]:
```suggestion
First of all, the dependency shall also be included in  in the ``CMakeLists.txt``:
```

File: docs/rst/tutorials/micro/custom_types/custom_types.rst [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+At this point, message definition in ``msg/MyCustomMessage.msg`` can now include types from the ``geometry_msgs`` package:
+
+.. code-block:: text
+
+  ...
+  int64 int64_test
+  uint64 uint64_test
+  geometry_msgs/Point32 point32_test
+
+And finally, the code can access this new member of in the custom type:
```
Comment (Acuadros95) [inline docs/rst/tutorials/micro/custom_types/custom_types.rst:139]:
```suggestion
And finally, the new member can be accessed in the custom type:
```

File: docs/rst/tutorials/micro/custom_types/custom_types.rst (hunk 4 above)
Comment (pablogs9) [reply docs/rst/tutorials/micro/custom_types/custom_types.rst]: eeff374fe8912d337d54725b469b98eb3b623c77

**Review by Acuadros95 — APPROVED** [review]

**Pull request merged**